linker/shared/src/test/scala/org/scalajs/linker/IRCheckerTest.scala

```diff
@@ -253,6 +253,38 @@ class IRCheckerTest {
     }
   }
 
+  @Test
+  def arrayOpsNull(): AsyncResult = await {
+    val classDefs = Seq(
+      mainTestClassDef(
+        Block(
+          ArraySelect(Null(), int(1))(NothingType),
+          ArrayLength(Null()),
+        )
+      )
+    )
+
+    testLinkNoIRError(classDefs, MainTestModuleInitializers)
+  }
+
+  @Test
+  def arrayAssignCovariant(): AsyncResult = await {
```

Comment:
Add a negative test here as well for an `Assign` to a primitive array type with the wrong type, to make sure we don't allow everything?